File: Python/Hamming Distance.py
class Solution(object):
    def hammingDistance(self, x, y):
        """
        :type x: int
        :type y: int
        :rtype: int
        """
        result = 0
        temp = x ^ y
        while temp > 0:
            result += temp - ((temp >> 1) << 1)
            temp >>= 1
        return result

File: cpp/Hamming Distance.cpp
class Solution {
public:
    int hammingDistance(int x, int y) {
        int counter = 0;
        int temp_num = x ^ y;
        while(temp_num > 0) {
            counter += temp_num & 1;
            temp_num >>= 1;
        }
        return counter;
    }
};
